changelog.d/4515.misc

```diff
@@ -0,0 +1 @@
+Voice recording mic button refactor with small animation tweaks in preparation for voice drafts
```

Comment:
wasn't 100% sure if the change log entry was needed as this is mostly refactoring (same as  https://github.com/vector-im/element-android/pull/4523)

Reply:
Always OK to have a changelog, if something goes wrong (I hope no!), we keep quickly see in the change history what have been done.